### PR TITLE
Job specific clothes storage slots now can carry bags

### PR DESCRIPTION
**Pull request by CursedBirb**
## About The Pull Request
Some time ago joined game as observer to chill a little. Noticed botanist having portable seed extractor on suit storage slot. After some thinking tried it in game and only botany had that option. Decided to add this possibility to other jobs. Since i knew nothing about byond coding i asked few people. With some help with setting git and how to code in byond changes were added. 
Also added tray sprite to mirro_belt file as it did not exist and pink belt was showing up.

## Why It's Good For The Game
Storage hungry jobs can now carry more things. Chemists can now make chems even if there is no power on station as can now carry both portable chem mixer and chembag. Virologist can help at medbay with medical belt on belt slot and viruses in bag on suit slot. Cook can now carry tools on suit storage or carry more food. Engineers can now use construction bag to its full potential as it can carry wires, cell, boards and some other things. Sci will most likely use it for more slimes extracts. Curator can carry more books now(Since curator does not have suit on its own it was added to treasure hunter set and normal wintercoat so it should work with everything that has wintercoat subtype). Very risky miners that uses miner wintercoat can also carry ore bag in suit storage slot. Did not added trashbag as was told that would lead to powergaming as bluespace one can carry 100 small items.

## Changelog

:cl:
qol: Job specific bags can now be carried on labcoats, wintercoats and apron suit storage slot
add: Added tray sprite to mirror_belt.dmi

/:cl:

### CODE REVIEW

File: code/modules/clothing/suits/jobs.dm [hunk 1]
```diff
@@ -51,7 +51,7 @@
 	inhand_icon_state = "apronchef"
 	blood_overlay_type = "armor"
 	body_parts_covered = CHEST|GROIN
-	allowed = list(/obj/item/kitchen, /obj/item/knife/kitchen)
+	allowed = list(/obj/item/kitchen, /obj/item/knife/kitchen, 	/obj/item/storage/bag/tray)
```
Comment (Fikou) [inline code/modules/clothing/suits/jobs.dm:54]: 2 spaces

File: code/modules/clothing/suits/jobs.dm [hunk 2]
```diff
@@ -39,7 +39,7 @@
 	inhand_icon_state = "chef"
 	permeability_coefficient = 0.5
 	body_parts_covered = CHEST|GROIN|ARMS
-	allowed = list(/obj/item/kitchen, /obj/item/knife/kitchen)
+	allowed = list(/obj/item/kitchen, /obj/item/knife/kitchen, 	/obj/item/storage/bag/tray)
```
Comment (Fikou) [inline code/modules/clothing/suits/jobs.dm:42]: ditto

File: code/modules/clothing/suits/labcoat.dm [hunk 3]
```diff
@@ -63,16 +63,34 @@
 	desc = "A suit that protects against minor chemical spills. Has an orange stripe on the shoulder."
 	icon_state = "labcoat_chem"
 
+/obj/item/clothing/suit/toggle/labcoat/chemist/Initialize(mapload)
+	. = ..()
+	allowed += list(
+		/obj/item/storage/bag/chemistry,
+	)
```
Comment (Fikou) [inline code/modules/clothing/suits/labcoat.dm:70]: why not just allowed += /obj/item/storage/bag/chemistry, same with the other things in this file

File: code/modules/clothing/suits/wintercoats.dm [hunk 4]
```diff
@@ -215,8 +216,15 @@
 	desc = "A lab-grade winter coat made with acid resistant polymers. For the enterprising chemist who was exiled to a frozen wasteland on the go."
 	icon_state = "coatchemistry"
 	inhand_icon_state = "coatchemistry"
+/obj/item/clothing/suit/hooded/wintercoat/medical/chemistry/Initialize(mapload)
```
Comment (Fikou) [inline code/modules/clothing/suits/wintercoats.dm:219]: put an enter here

File: code/modules/clothing/suits/wintercoats.dm [hunk 5]
```diff
@@ -227,8 +235,17 @@
 	desc = "A white winter coat with green markings. Warm, but wont fight off the common cold or any other disease. Might make people stand far away from you in the hallway. The zipper tab looks like an oversized bacteriophage."
 	icon_state = "coatviro"
 	inhand_icon_state = "coatviro"
+/obj/item/clothing/suit/hooded/wintercoat/medical/viro/Initialize(mapload)
```
Comment (Fikou) [inline code/modules/clothing/suits/wintercoats.dm:238]: and here

File: code/modules/clothing/suits/wintercoats.dm [hunk 6]
```diff
@@ -215,8 +216,15 @@
...
 	inhand_icon_state = "coatchemistry"
+/obj/item/clothing/suit/hooded/wintercoat/medical/chemistry/Initialize(mapload)
+	. = ..()
+	allowed += list(
+		/obj/item/storage/bag/chemistry,
+	)
+
 	hoodtype = /obj/item/clothing/head/hooded/winterhood/medical/chemistry
 
+
```
Comment (Fikou) [inline code/modules/clothing/suits/wintercoats.dm:227]: why

File: code/modules/clothing/suits/wintercoats.dm [hunk 7]
```diff
@@ -227,8 +235,17 @@
...
+	. = ..()
+	allowed += list(
+		/obj/item/storage/bag/bio,
+
+	)
+
 	hoodtype = /obj/item/clothing/head/hooded/winterhood/medical/viro
 
+
+
```
Comment (Fikou) [inline code/modules/clothing/suits/wintercoats.dm:248]: why

File: code/modules/clothing/suits/wintercoats.dm [hunk 8]
```diff
@@ -262,6 +279,8 @@
 		/obj/item/reagent_containers/pill,
 		/obj/item/reagent_containers/syringe,
 		/obj/item/storage/pill_bottle,
+		/obj/item/storage/bag/bio,
+
```
Comment (Fikou) [inline code/modules/clothing/suits/wintercoats.dm:283]: remove this newline

File: code/modules/clothing/suits/wintercoats.dm [hunk 9]
```diff
@@ -227,8 +235,17 @@
...
 	icon_state = "coatviro"
 	inhand_icon_state = "coatviro"
+/obj/item/clothing/suit/hooded/wintercoat/medical/viro/Initialize(mapload)
+	. = ..()
+	allowed += list(
+		/obj/item/storage/bag/bio,
+
+	)
+
 	hoodtype = /obj/item/clothing/head/hooded/winterhood/medical/viro
```
Comment (Fikou) [inline code/modules/clothing/suits/wintercoats.dm:245]: you have object definition stuff in your initialize proc

File: code/modules/clothing/suits/jobs.dm (hunk 1 above)
Comment (Amrabol) [reply code/modules/clothing/suits/jobs.dm:54]: Fixed, but will post it later as I have no Ethernet currently and writing from phone

File: code/modules/clothing/suits/jobs.dm (hunk 2 above)
Comment (Amrabol) [reply code/modules/clothing/suits/jobs.dm:42]: Fixed, but will post it later as I have no Ethernet currently and writing from phone

File: code/modules/clothing/suits/wintercoats.dm (hunk 4 above)
Comment (Amrabol) [reply code/modules/clothing/suits/wintercoats.dm:219]: Fixed, but will post it later as I have no Ethernet currently and writing from phone

File: code/modules/clothing/suits/wintercoats.dm (hunk 5 above)
Comment (Amrabol) [reply code/modules/clothing/suits/wintercoats.dm:238]: Fixed, but will post it later as I have no Ethernet currently and writing from phone

File: code/modules/clothing/suits/wintercoats.dm (hunk 6 above)
Comment (Amrabol) [reply code/modules/clothing/suits/wintercoats.dm:227]: Fixed, but will post it later as I have no Ethernet currently and writing from phone

File: code/modules/clothing/suits/wintercoats.dm (hunk 7 above)
Comment (Amrabol) [reply code/modules/clothing/suits/wintercoats.dm:248]: Fixed, but will post it later as I have no Ethernet currently and writing from phone

File: code/modules/clothing/suits/wintercoats.dm (hunk 8 above)
Comment (Amrabol) [reply code/modules/clothing/suits/wintercoats.dm:283]: Fixed, but will post it later as I have no Ethernet currently and writing from phone

File: code/modules/clothing/suits/wintercoats.dm (hunk 9 above)
Comment (Amrabol) [reply code/modules/clothing/suits/wintercoats.dm:245]: Fixed, but will post it later as I have no Ethernet currently and writing from phone

File: code/modules/clothing/suits/labcoat.dm (hunk 3 above)
Comment (Amrabol) [reply code/modules/clothing/suits/labcoat.dm:70]: Tried to do it that way but keep getting error with: "got '+-', expected one of: '/', contents, newline, identifier". So decided to do it like it was done to cmo winter coat

**Comment by Amrabol** [comment]
On request added construction bag to hazard vest and ore bag to explorer suit. Will push them once I get Ethernet back

File: code/modules/clothing/suits/labcoat.dm (hunk 3 above)
Comment (Ghommie) [reply code/modules/clothing/suits/labcoat.dm:70]: You probably got the syntax wrong. Here's a suggestion to help you:
```suggestion
	allowed += /obj/item/storage/bag/chemistry
```

File: code/modules/clothing/suits/labcoat.dm [hunk 10]
```diff
@@ -63,16 +63,34 @@
...
 /obj/item/clothing/suit/toggle/labcoat/virologist
 	name = "virologist labcoat"
 	desc = "A suit that protects against minor chemical spills. Has a green stripe on the shoulder."
 	icon_state = "labcoat_vir"
 
+/obj/item/clothing/suit/toggle/labcoat/virologist/Initialize(mapload)
+	. = ..()
+	allowed += list(
+		/obj/item/storage/bag/bio,
+	)
```
Comment (Ghommie) [inline code/modules/clothing/suits/labcoat.dm:81]:
```suggestion
	allowed += /obj/item/storage/bag/bio
```

File: code/modules/clothing/suits/labcoat.dm [hunk 11]
```diff
@@ -63,16 +63,34 @@
...
 /obj/item/clothing/suit/toggle/labcoat/science
 	name = "scientist labcoat"
 	desc = "A suit that protects against minor chemical spills. Has a purple stripe on the shoulder."
 	icon_state = "labcoat_sci"
 
+/obj/item/clothing/suit/toggle/labcoat/science/Initialize(mapload)
+	. = ..()
+	allowed += list(
+		/obj/item/storage/bag/bio,
+	)
```
Comment (Ghommie) [inline code/modules/clothing/suits/labcoat.dm:92]:
```suggestion
	allowed += /obj/item/storage/bag/bio
```

File: code/modules/clothing/suits/wintercoats.dm [hunk 12]
```diff
@@ -215,8 +216,15 @@
 	desc = "A lab-grade winter coat made with acid resistant polymers. For the enterprising chemist who was exiled to a frozen wasteland on the go."
 	icon_state = "coatchemistry"
 	inhand_icon_state = "coatchemistry"
+/obj/item/clothing/suit/hooded/wintercoat/medical/chemistry/Initialize(mapload)
+	. = ..()
+	allowed += list(
+		/obj/item/storage/bag/chemistry,
+	)
```
Comment (Ghommie) [inline code/modules/clothing/suits/wintercoats.dm:223]:
```suggestion
	allowed += /obj/item/storage/bag/chemistry
```

File: code/modules/clothing/suits/wintercoats.dm [hunk 13]
```diff
@@ -227,8 +235,17 @@
 	desc = "A white winter coat with green markings. Warm, but wont fight off the common cold or any other disease. Might make people stand far away from you in the hallway. The zipper tab looks like an oversized bacteriophage."
 	icon_state = "coatviro"
 	inhand_icon_state = "coatviro"
+/obj/item/clothing/suit/hooded/wintercoat/medical/viro/Initialize(mapload)
+	. = ..()
+	allowed += list(
+		/obj/item/storage/bag/bio,
+
+	)
```
Comment (Ghommie) [inline code/modules/clothing/suits/wintercoats.dm:243]:
```suggestion
	allowed += /obj/item/storage/bag/bio
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/modules/clothing/suits/wintercoats.dm (hunk 13 above)
Comment (Amrabol) [reply code/modules/clothing/suits/wintercoats.dm:243]: Will change and push them once I get connection back

File: code/modules/clothing/suits/wintercoats.dm (hunk 12 above)
Comment (Amrabol) [reply code/modules/clothing/suits/wintercoats.dm:223]: Will change and push them once I get connection back

File: code/modules/clothing/suits/labcoat.dm (hunk 11 above)
Comment (Amrabol) [reply code/modules/clothing/suits/labcoat.dm:92]: Will change and push them once I get connection back

File: code/modules/clothing/suits/labcoat.dm (hunk 10 above)
Comment (Amrabol) [reply code/modules/clothing/suits/labcoat.dm:81]: Will change and push them once I get connection back

**Comment by Amrabol** [comment]
Looks like new method is not working. Should i go back to previous method?

**Comment by Amrabol** [comment]
Or i might have done it wrong. I will try to change it again

**Comment by Ghommie** [comment]
Oh! so that's why it's throwing compile errors. You are doing these operations in the definition of the object and not on `object/Initialize()` as I previously thought. 
Sorry for suggesting that earlier. You should actually move these `allowed += bag/type/path` lines from the object definition to their `Initialize()`.

**Comment by Amrabol** [comment]
> Oh! so that's why it's throwing compile errors. You are doing these operations in the definition of the object and not on `object/Initialize()` as I previously thought. Sorry for suggesting that earlier. You should actually move these `allowed += bag/type/path` lines from the object definition to their `Initialize()`. EDIT: I see you're already doing that, gj.

At this point i decided to go back and try to make old version better. Also i start to understand why people put weird names for commits as i slowly started to regret trying to do this pr

**Comment by Amrabol** [comment]
I need to make change again as this time it does not work for both viro and chemist winter coat. Also once again I lose connection

**Comment by Amrabol** [comment]
Turns out i copied from part of code for both viro and chem so i had to make some changes and this time it worked nice. Testing was done

File: code/modules/clothing/suits/labcoat.dm [hunk 14]
```diff
@@ -63,16 +63,28 @@
...
+	allowed +=	/obj/item/storage/bag/bio
+
 /obj/item/clothing/suit/toggle/labcoat/science
 	name = "scientist labcoat"
 	desc = "A suit that protects against minor chemical spills. Has a purple stripe on the shoulder."
 	icon_state = "labcoat_sci"
 
+/obj/item/clothing/suit/toggle/labcoat/science/Initialize(mapload)
+	. = ..()
+	allowed +=	/obj/item/storage/bag/bio
```
Comment (Fikou) [inline code/modules/clothing/suits/labcoat.dm:86]: remove the double spaces after += other than that looks good 👍

Comment (Amrabol) [reply code/modules/clothing/suits/labcoat.dm]: Fixed

**Review by Kylerace — APPROVED** [review]
code is good

**Comment by Krysonism** [comment]
Carrying a tray on the belt or suit makes no sense, but if maintainers insist we must have this the sprite is okay.

**Pull request merged**